### ISSUE TITLE
Disallow UID as a predicate name

### ISSUE THREAD

**Issue by harshil-goel** [dgraph-io/dgraph#4218]
### What version of Dgraph are you using?
v1.1.0

### Have you tried reproducing the issue with the latest release?
Yes

### What is the hardware spec (RAM, OS)?
32gb RAM, Ubuntu 19.04

### Steps to reproduce the issue (command/config used to run Dgraph).
Add `<uid>: string .` as a schema works.

### Expected behavior and actual result.
Expected behavior: An error saying UID is a reserved word and cannot be used as a predicated. 
Currently, you can set `<uid>: string .` and then  
```
{
  set {
            _:alice <uid> "Hi" .
            _:alice <dgraph.Type> "Person" .
   }
}

{
   q(func: type(Person)) {
             uid
    }
}
```
would result in 
```
{
   "q": [
        {
          "uid": "0xa8"
        }, 
     ] 
}
```

**Comment by martinmr** [comment]
If you added a mutation with a uid predicate but no schema, dgraph would allow it and create a default predicate in the schema. This should also be disallowed.

**Comment by martinmr** [comment]
Fixed by #4219
